sqllogictest/src/parser.rs

```diff
@@ -417,6 +417,12 @@ fn parse_inner(loc: &Location, script: &str) -> Result<Vec<Record>, ParseError>
             records.push(Record::Comment(comments));
         }
 
+        // Consider a case: the last line is a comment
+        // Then line will start with '#', but it has been process at above
+        if line.starts_with('#') {
+            continue;
```

Comment:
I guess it actually should be `break`? It took me a while to understand why previously it is wrong. 🤣 The real problem is we didn't consider the case where iterator ends at:

```rust
for (num_, line_) in lines.by_ref() 
```

So what about refactor the `for` loop to some other form using `lines.next()`, and `break`s there if it returns `None`.